app/models/internal/impl/ChromeScreenshotReportSource.java

```diff
@@ -0,0 +1,203 @@
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final ChromeScreenshotReportSource that = (ChromeScreenshotReportSource) o;
+        return _ignoreCertificateErrors == that._ignoreCertificateErrors
+                && Objects.equals(getId(), that.getId())
+                && Objects.equals(getUrl(), that.getUrl())
+                && Objects.equals(getTitle(), that.getTitle())
+                && Objects.equals(getTriggeringEventName(), that.getTriggeringEventName());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId(), getUrl(), getTitle(), _ignoreCertificateErrors, getTriggeringEventName());
+    }
+
+    /**
+     * Builder implementation that constructs {@link ChromeScreenshotReportSource}.
+     */
+    public static final class Builder extends OvalBuilder<ChromeScreenshotReportSource> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(ChromeScreenshotReportSource::new);
+            _ignoreCertificateErrors = false;
+        }
+
+        /**
+         * The source id. Required. Cannot be null or empty.
+         * @param id The id.
+         * @return This instance of {@code Builder}
+         */
+        public Builder setId(final UUID id) {
+            _id = id;
+            return this;
+        }
+
+        /**
+         * The source URL. Required. Cannot be null or empty.
+         * @param url The URL.
+         * @return This instance of {@code Builder}
+         */
+        public Builder setUrl(final String url) {
+            _url = url;
+            return this;
+        }
+
+        /**
+         * The source title. Required. Cannot be null or empty.
+         * @param title The title.
+         * @return This instance of {@code Builder}
+         */
+        public Builder setTitle(final String title) {
+            _title = title;
+            return this;
+        }
+
+        /**
+         * The triggering event name for the source. Required. Cannot be null or empty.
+         * @param triggeringEventName The event name.
+         * @return This instance of {@code Builder}
+         */
+        public Builder setTriggeringEventName(final String triggeringEventName) {
+            _triggeringEventName = triggeringEventName;
+            return this;
+        }
+
+        /**
+         * Determine if the source should ignore certificate errors. Defaults to false.
+         * @param ignoreCertificateErrors Whether this source should ignore certificate errors.
+         * @return This instance of {@code Builder}
+         */
+        public Builder setIgnoreCertificateErrors(final boolean ignoreCertificateErrors) {
+            _ignoreCertificateErrors = ignoreCertificateErrors;
+            return this;
+        }
+
+        @NotNull
+        @NotEmpty
+        private UUID _id;
+        @NotNull
+        @NotEmpty
+        @AssertURL
```

Comment:
Probably better to type this as a `URI`. We may need to assert some properties of the URI; e.g. that it's protocol is either `http` or `https`.